master/pkg/model/metrics.go

```diff
@@ -46,3 +57,38 @@ func (t MetricType) Validate() error {
 	}
 	return nil
 }
+
+// MetricIdentifier packages metric type and name together.
+type MetricIdentifier struct {
+	Type MetricType
+	Name metricName
```

Comment:
Hm, seems strange to have an exported property that's of an unexported type. Did you mean `ValidateMetricName` could be unexported, @stoksc?

Reply:
I think that's what he meant. I made metricName protected because usage of it as a type is not fully clear to me yet so I didn't want to export it just yet. Ideally when it's exported all our metric name references elsewhere are also updated to this type which would be even more out of scope here IMO

Reply:
Reasonable!